hw/ip/spi_device/doc/registers.md

```diff
@@ -0,0 +1,1791 @@
+| spi_device.[`CONTROL`](#control)                         | 0x10     |        4 | Control register                                        |
+| spi_device.[`CFG`](#cfg)                                 | 0x14     |        4 | Configuration Register                                  |
+| spi_device.[`FIFO_LEVEL`](#fifo_level)                   | 0x18     |        4 | RX/ TX FIFO levels.                                     |
+| spi_device.[`ASYNC_FIFO_LEVEL`](#async_fifo_level)       | 0x1c     |        4 | RX/ TX Async FIFO levels between main clk and spi clock |
+| spi_device.[`STATUS`](#status)                           | 0x20     |        4 | SPI Device status register                              |
+| spi_device.[`RXF_PTR`](#rxf_ptr)                         | 0x24     |        4 | Receiver FIFO (SRAM) pointers                           |
+| spi_device.[`TXF_PTR`](#txf_ptr)                         | 0x28     |        4 | Transmitter FIFO (SRAM) pointers                        |
+| spi_device.[`RXF_ADDR`](#rxf_addr)                       | 0x2c     |        4 | Receiver FIFO (SRAM) Addresses                          |
+| spi_device.[`TXF_ADDR`](#txf_addr)                       | 0x30     |        4 | Transmitter FIFO (SRAM) Addresses                       |
+| spi_device.[`INTERCEPT_EN`](#intercept_en)               | 0x34     |        4 | Intercept Passthrough datapath.                         |
+| spi_device.[`LAST_READ_ADDR`](#last_read_addr)           | 0x38     |        4 | Last Read Address                                       |
+| spi_device.[`FLASH_STATUS`](#flash_status)               | 0x3c     |        4 | SPI Flash Status register.                              |
+| spi_device.[`JEDEC_CC`](#jedec_cc)                       | 0x40     |        4 | JEDEC Continuation Code configuration register.         |
+| spi_device.[`JEDEC_ID`](#jedec_id)                       | 0x44     |        4 | JEDEC ID register.                                      |
+| spi_device.[`READ_THRESHOLD`](#read_threshold)           | 0x48     |        4 | Read Buffer threshold register.                         |
+| spi_device.[`MAILBOX_ADDR`](#mailbox_addr)               | 0x4c     |        4 | Mailbox Base address register.                          |
+| spi_device.[`UPLOAD_STATUS`](#upload_status)             | 0x50     |        4 | Upload module status register.                          |
+| spi_device.[`UPLOAD_STATUS2`](#upload_status2)           | 0x54     |        4 | Upload module status 2 register.                        |
+| spi_device.[`UPLOAD_CMDFIFO`](#upload_cmdfifo)           | 0x58     |        4 | Command Fifo Read Port.                                 |
+| spi_device.[`UPLOAD_ADDRFIFO`](#upload_addrfifo)         | 0x5c     |        4 | Address Fifo Read Port.                                 |
+| spi_device.[`CMD_FILTER_0`](#CMD_FILTER_0)               | 0x60     |        4 | Command Filter                                          |
```

Comment:
The fragments here need to be lowercase, as the url fragments are case-sensitive.